core/shared/src/main/scala/coursier/util/Print.scala

```diff
@@ -80,6 +80,17 @@ object Print {
     reverse: Boolean,
     colors: Boolean
   ): String = {
+    val backwardCompatibleDependencies = if (reverse) resolution.dependencies.toSeq else roots
+    finalDependencyTree(backwardCompatibleDependencies, resolution, printExclusions, reverse, colors)
```

Comment:
I am fine with changing the behavior for cli as long as the new behavior is correct, i.e. the root nodes of the printed tree should be the leaf nodes of the resolve, and then there is no need for `finalDependencyTree`.

The reason I am fine with the change is that I don't think anyone is trying to parse the cli tree output, as normally I'd lean toward the deprecation route.

I'd also like to see some tests on this.

Maybe @alexarchambault has more thoughts.

Reply:
I’m not sure but I think that if we want to test it, we should return some data structure of a tree or sequence of trees from a new method (just to keep the API backward compatible) and have another render method that will render it to the CLI. 

Now I’m not sure how to “mock” a resolution with dependencies graph..  

I think that if you want to change the behavior of the CLI then the command should be more like:

“coursier resolve some:module:version -root another:module:version”

This will render a reverse graph from another module version to some module version...

Reply:
Since the change is in `core/`, it will affect how cli runs as well.

In cli `resolve -T ..` means printing the reverse tree, so it should print from all leaf nodes all the way to the root nodes`, in this case
```
# no <-root another:module:version> needed here
$ resolve -T junit:junit:4.12
└─ org.hamcrest:hamcrest-core:1.3
   └─ junit:junit:4.12
```

Also I think it would be beneficial to keep the behavior consistent between cli and sbt. If you are only interested to a sub tree from a particular root from sbt, I think it's fine to add separate functionality to it without changing the existing code.

You may be able to look around and see how tests can be added (I am not entirely familiar with this portion yet, so I'd need to that do too), but in general tests would be required for changes in `core/`

Reply:
I’m sorry but I’m not sure how printing the reversed dependencies graph for a single dependency is benificial for someone. Can you give a use case for that?

Reply:
No, but I thought you were by proposing `-root another:module:version`

My point is that the CLI output needs to be correct, i.e. starting from all the leaf nodes, and ideally consistent with sbt output.

Reply:
@wisechengyi so can you summarize what you want to happen in every command in the CLI and API and I’ll just do it in the next 24 hours? :)

Thanks.